__tests__/e2e/AddFriendsStateFlow.test.tsx

```diff
@@ -0,0 +1,418 @@
+import {
+  render,
+  fireEvent,
+  waitFor,
+  cleanup,
+  act,
+} from "@testing-library/react-native";
+import { NavigationContainer } from "@react-navigation/native";
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import FirestoreCtrl, {
+  DBUser,
+  DBChallenge,
+  DBChallengeDescription,
+  DBGroup,
+  DBComment,
+} from "@/src/models/firebase/FirestoreCtrl";
+import HomeScreen from "@/src/views/home/home_screen";
+import FriendsScreen from "@/src/views/friends/friends_screen";
+import MaximizeScreen from "@/src/views/home/maximize_screen";
+
+const Stack = createNativeStackNavigator();
+
+// Mock FirestoreCtrl
+jest.mock("@/src/models/firebase/FirestoreCtrl", () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      // Mock functions used in home screen
+      getGroupsByUserId: jest.fn((id) => {
+        if (id === "123") {
+          return new Promise<DBGroup[]>((resolve) => {
+            resolve([mockTesterGroup]);
+          });
+        } else if (id === "456") {
+          return new Promise<DBGroup[]>((resolve) => {
+            resolve([mockTesterFriendGroup]);
+          });
+        }
+      }),
+      getChallengeDescription: jest.fn((id) => {
+        return mockCurrentChallenge;
+      }),
+      getPostsByChallengeTitle: jest.fn((title) => {
+        return new Promise<DBChallenge[]>((resolve) => {
+          resolve([mockPostTesterFriend]);
+        });
+      }),
+
+      // Mock functions used in friends screen
+      getAllUsers: jest.fn(() => {
+        return [mockTester, mockTesterFriend];
+      }),
+      addFriend: jest.fn((uid, friendUid) => {
+        if (uid === "123" && friendUid === "456") {
+          mockTester.friends.push(friendUid);
+        } else if (uid === "456" && friendUid === "123") {
+          mockTesterFriend.friends.push(friendUid);
+        }
+      }),
+      acceptFriend: jest.fn((uid, friendUid) => {
+        if (uid === "456" && friendUid === "123") {
+          mockTesterFriend.friends.push(friendUid);
+        } else if (uid === "123" && friendUid === "456") {
+          mockTester.friends.push(friendUid);
+        }
+      }),
+      getFriendSuggestions: jest.fn((uid) => {
+        if (uid === "123") {
+          return [];
+        } else if (uid === "456") {
+          return [mockTester];
+        }
+      }),
+      getFriends: jest.fn((uid) => {}),
+      getFriendRequests: jest.fn((uid) => {
+        if (uid === "456") {
+          return [mockTester];
+        } else return [];
+      }),
+      isFriend: jest.fn((uid, friendUid) => {
+        if (!isRequestSent) {
+          return false;
+        } else {
+          if (mockTester.friends.includes(friendUid)) {
+            return true;
+          }
+        }
+      }),
+      isRequested: jest.fn((uid, friendUid) => {
+        if (!isRequestSent) {
+          return false;
+        } else {
+          if (uid === "456") {
+            return true;
+          }
```

Comment:
The mock isn't fully complete as this method has some edge-cases that return nothing : `isRequestSent=true` and `uid != 456`

Reply:
Well mocking the full function is not necessary here, and otherwise the mocked function does nothing.